### DISCUSSION TITLE
MUMPS options with Ipopt (Matlab)

### DISCUSSION THREAD

**Discussion by federicoc-wae** [casadi/casadi#3925] [Q&A]
Hi,

Is it possible to change the linear solver MUMPS options when solving NLP with Ipopt?
I am using Ipopt version 3.14.11, running with linear solver MUMPS 5.4.1.

I am using the Opti class in Matlab.

Can anybody help me with an example code?

Many thanks

**Comment by Lars-DHondt-KUL** [comment]
You can set set some MUMPS options via the [ipopt options](https://coin-or.github.io/Ipopt/OPTIONS.html#OPT_Mumps_Linear_Solver).
```
optionssol.ipopt.mumps_pivot_order = 5; % metis
optionssol.ipopt.mumps_scaling = 0; % none
opti.solver('ipopt',optionssol);
```
You can look up the ICNTL index in the MUMPS documentation to know what each value means.
